src/taskmanager.cpp

```diff
@@ -430,11 +435,42 @@ void TaskManager::findMatchingArchives(QString jobPrefix)
     emit matchingArchives(matching);
 }
 
+bool TaskManager::waitForOnline()
+{
+    for(int i = 0; i < NETWORK_UP_ATTEMPTS; i++)
+    {
+        QTcpSocket sock;
+        sock.connectToHost(NETWORK_HOST, NETWORK_PORT);
+        // Multiple calls to waitForConnected() do not accumulate time, so we
+        // use ::sleep() below.
+        if(sock.waitForConnected(500))
+        {
+            sock.disconnectFromHost();
+            return true;
+        }
+        // Wait a bit, then try again
+        if(i < NETWORK_UP_ATTEMPTS - 1)
```

Comment:
What's up with this check also? Do you really want to attempt 1 less then what the actual define hints at?

Reply:
Yes, we want to wait for 1 less attempt than we check.  For example,

```
check 0
wait 0
check 1
wait 1
check 2
```

There's no point having a `wait 2` if we don't check again after that.